typescript/web/src/components/import-button/import-images-modal/modal-dropzone/file-statuses.tsx

```diff
@@ -1,68 +1,147 @@
+      warnings.map((warning) => (
+        <Box
+          pr="2"
+          overflow="hidden"
+          textOverflow="ellipsis"
+          fontSize="xs"
+          color={mode("red.500", "red.300")}
+        >
+          {warning}
+        </Box>
+      ))}
+  </>
+);
+
+const FileStatus = ({
+  droppedFile,
+  index,
+  fileUploadInfo,
+}: FileStatusProps) => {
+  const warnings = fileUploadInfo?.warnings;
+  const FileStatusIcon = fileStatusIcon(
+    droppedFile.file,
+    !isEmpty(droppedFile.errors)
+  );
+  return (
+    <>
+      <Flex
+        key={droppedFile.file.path}
+        w="100%"
+        alignItems="center"
+        p="2"
+        bg={index % 2 === 0 ? mode("gray.50", "gray.700") : "inherit"}
+      >
+        <Box flex={0} pr="2">
+          <FileStatusIcon />
+        </Box>
+        <FilePath droppedFile={droppedFile} />
+        <ImportStatus
+          droppedFile={droppedFile}
+          fileUploadInfo={fileUploadInfo}
+        />
+      </Flex>
+      <Warnings warnings={warnings} />
+    </>
+  );
+};
+
 export const FilesStatuses = ({
   files,
-  fileUploadStatuses,
+  fileUploadInfoRecord,
 }: {
   files: Array<DroppedFile>;
-  fileUploadStatuses: UploadStatuses;
+  fileUploadInfoRecord: UploadInfoRecord;
 }) => (
   <Flex direction="column" height="100%">
     <Box p="2" bg={mode("gray.200", "gray.600")} borderTopRadius="md" w="100%">
       <Text>
         Completed{" "}
         {
-          Object.entries(fileUploadStatuses).filter(
-            (entry) => entry[1] === true
+          Object.entries(fileUploadInfoRecord).filter(
+            (entry) => entry[1].status === true
           ).length
         }{" "}
         of {files.filter((file) => isEmpty(file.errors)).length} items
       </Text>
     </Box>
     <Flex direction="column" overflowY="auto" width="100%" height="100%">
-      {files.map(({ file, errors }, index) => (
-        <Flex
-          key={file.path}
-          w="100%"
-          alignItems="center"
-          p="2"
-          bg={index % 2 === 0 ? mode("gray.50", "gray.700") : "inherit"}
-        >
-          <Box flex={0} pr="2">
-            {isEmpty(errors) ? <RiImageLine /> : <RiFile3Line />}
-          </Box>
-          <Box
-            pr="2"
-            flexGrow={1}
-            flexShrink={1}
-            overflow="hidden"
-            textOverflow="ellipsis"
-            whiteSpace="nowrap"
-          >
-            {file.path}
-          </Box>
-          <Box
-            whiteSpace="nowrap"
-            flex={0}
-            color={mode("gray.400", "gray.600")}
-            fontSize="md"
-            textAlign="right"
-          >
-            {isEmpty(errors) ? (
-              <ImportProgress
-                status={fileUploadStatuses[file.path ?? file.name]}
-              />
-            ) : (
-              <ImportError errors={errors} />
-            )}
-          </Box>
-        </Flex>
+      {files.map((droppedFile, index) => (
+        <FileStatus
```

Comment:
We need a key here too (⚠️ [_using index will not be possible_](https://medium.com/geekculture/reactjs-why-index-as-a-key-is-an-anti-pattern-4b9dc6ef0067)).